packages/java-parser/src/productions/expressions.js

```diff
@@ -166,6 +169,8 @@ function defineRules($, t) {
           }
         },
         {
+          // Remove ambiguity with first OR option
+          GATE: () => !tokenMatcher($.LA(1).tokenType, t.Instanceof),
```

Comment:
is the ambiguity is resolved by the order (instanceOf is first) you don't need a GATE
rather you can ignore the ambiguities in this specific ALT